primitives/core/src/testing.rs

```diff
@@ -241,15 +242,15 @@ impl crate::traits::BareCryptoStore for KeyStore {
 		}
 	}
 
-	fn sr25519_vrf_sign(
+	async fn sr25519_vrf_sign(
 		&self,
 		key_type: KeyTypeId,
 		public: &sr25519::Public,
 		transcript_data: VRFTranscriptData,
 	) -> Result<VRFSignature, Error> {
 		let transcript = make_transcript(transcript_data);
 		let pair = self.sr25519_key_pair(key_type, public)
-			.ok_or_else(|| Error::PairNotFound("Not found".to_owned()))?;
+			.ok_or(Error::PairNotFound("Not found".to_owned()))?;
```

Comment:
why making that allocation when there's no error?
```suggestion
			.ok_or_else(|| Error::PairNotFound("Not found".to_owned()))?;
```